tests/checkPot.py

```diff
@@ -120,8 +138,9 @@ def checkPot(fileName):
 				sourceLines = []
 				context = ""
 				continue
-			if line == 'msgid ""':
+			if line == 'msgid ""' and not passedHeader:
 				# Header.
+				passedHeader = True

```

Comment:
Shouldn't `passedHeader` be reset to `False` somewhere in the `for line in pot` loop? 

Would you mind adding a few unit tests for this code? Happy to accept in a subsequent PR? Or raise an issue.

Reply:
> Shouldn't `passedHeader` be reset to `False` somewhere in the `for line in pot` loop?

Surely not.
The block we are skipping here occurs only once, at the top of the file:
```
#, fuzzy
msgid ""
msgstr ""
```


> Would you mind adding a few unit tests for this code? Happy to accept in a subsequent PR? Or raise an issue.

Adding a unit test for this global test? Sure! (I pretty much liked Inception...)
Good exercise for me as I haven't contributed a new unit test yet.
Probably for a subsequent PR though, as I guess merging this one is more urgent than preventing a subsequent change from breaking that code.

Reply:
Ok, subsequent PR is fine with me. Part of my motivation for tests was to make it easier to understand how this works in small pieces while reviewing this PR. I'll go read the code more carefully instead, it must work differently to how I imagine. But regardless, tests will make future modifications much easier. I'll probably forget how this works by the time we need to modify it again.

Reply:
POT files contain an initial empty message:
```
msgid ""
msgstr ""
```
The `checkPot` test should not check for a translators comment there.
It should nevertheless check for a translators comment above all subsequent messages.

Short translatable strings are extracted by gettext like so:
```
#. Translators: Reported when single letter navigation in browse mode is turned on.
#: browseMode.py:377
msgid "Single letter navigation on"
msgstr ""
```

When a translatable string exceeds a certain length, gettext extract it as:
```
#. Translators: the description for the toggleSingleLetterNavigation command in browse mode.
#: browseMode.py:379
msgid ""
"Toggles single letter navigation on and off. When on, single letter keys in "
"browse mode jump to various kinds of elements on the page. When off, these "
"keys are passed to the application"
msgstr ""
```
That is, there is a `msgid ""` at the start.

Before my proposed change, `checkPot` was mistaken these `msgid ""` for the header it should ignore. Thus, any lengthy translatable string was not checked for the presence of a comment.
My proposed change flags the first time we encounter `msgid ""` so that we no longer ignore it later on.